st2client/st2client/commands/action.py

```diff
@@ -1505,6 +1520,16 @@ def tail_execution(cls, execution_manager, stream_manager, execution, output_typ
                 # An execution is considered a child execution if it has parent execution id
                 is_child_execution = bool(task_parent_execution_id)
 
+                # Ignore executions which are not part of the execution we are tailing
+                if is_child_execution:
```

Comment:
Ideally we could eventually perform this smart "belongs to parent execution X" filtering on the event stream API side.

It would reduce workload and traffic on the client and it could also be used by the WebUI. For now this should work though.